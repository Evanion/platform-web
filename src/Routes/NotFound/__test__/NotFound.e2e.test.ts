import puppeteer, { Browser, Page } from 'puppeteer';

let browser: Browser;
let page: Page;

describe('NotFound Route E2E', () => {
  beforeEach(async () => {
    browser = await puppeteer.launch();
    page = await browser.newPage();

    page.emulate({
      viewport: {
        width: 1920,
        height: 1080,
      },
      userAgent: '',
    });
  });

  afterEach(async () => {
    await browser.close();
  });

  it("should be displayed if the requested route doesn't exist", async () => {
    await page.goto('http://localhost:3000/you-dont-exist');
    await page.waitForSelector('#root > div');

    const html = await page.$eval('#root > div', e => e.innerHTML);

    // @ts-ignore
    expect(html).toMatchSnapshot();
  });
});
