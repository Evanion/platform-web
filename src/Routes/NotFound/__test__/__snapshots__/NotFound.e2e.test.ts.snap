// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Route E2E should be displayed if the requested route doesn't exist 1`] = `""`;
